6_nodejs/week2/app.js

```diff
@@ -0,0 +1,78 @@
+import {default as express} from "express";
+import fs from "fs";
+
+const app = express();
+const port = process.env.PORT || 3000;
+const path = "./documents.json";
+const documents = JSON.parse(fs.readFileSync(path))
+
+app.use(express.json());
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+//GET /search
+app.get("/search", (req, res) => {
+   const query = req.query.q;
+    if (!query) {
+            res.json(documents);
+    } else {
+        const filteredDocuments = documents.filter((doc) => {
+            return Object.values(doc).some((value) =>
+              value.toString().includes(query)
+            );
+          });
+          if (filteredDocuments) {
+          res.json(filteredDocuments);
+        } else {
+          res.send("Unable to found");
```

Comment:
you might want to keep the endpoint response consistent - so if for a certain query the response is a Json with an array of objects, then if there is nothing found for a different query the endpoint should return a json with empty array, not a string instead of json. That way you keep the type of data that endpoint return the same - it will always be a json and that will be easier to handle for anyone who is using your API.